### PR TITLE
AB2-591 Avoid adding click listeners to CC ui elements when data is n…

### PR DESCRIPTION
**Pull request by jayeshsolanki93**
…ot ready

### CODE REVIEW

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/controlcenter/OverviewFragment.java [hunk 1]
```diff
@@ -357,7 +357,7 @@ public void refreshUI() {
 
     @Override
     public void onClick(View v) {
-        if (controlCenterSettingsData == null) {
+        if (controlCenterSettingsData == null || controlCenterSettingsData.getBundle("data") == null) {
```
Comment (moaz-cliqz) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/controlcenter/OverviewFragment.java]: I would prefer to use || GeckoBundleUtils.safeGetBundle(controlCenterSettingsData,"data") == null )

Comment (ravjit-cliqz) [reply mozilla-release/mobile/android/base/java/org/mozilla/gecko/controlcenter/OverviewFragment.java]: It will never be null if you use `safeGetBundle`.

Comment (ravjit-cliqz) [reply mozilla-release/mobile/android/base/java/org/mozilla/gecko/controlcenter/OverviewFragment.java]: Best would be to replace all the `get` calls to `safeGet`. There are still some palces in the code where we didnt change to `safeGet`. That would prevent any crash which is happening.

File: mozilla-release/mobile/android/base/java/org/mozilla/gecko/controlcenter/OverviewFragment.java [hunk 2]
```diff
@@ -357,7 +357,9 @@ public void refreshUI() {
 
     @Override
     public void onClick(View v) {
-        if (controlCenterSettingsData == null) {
+
+        if (controlCenterSettingsData == null ||
+                GeckoBundleUtils.safeGetBundle(controlCenterSettingsData, "data") == null) {
```
Comment (moaz-cliqz) [inline mozilla-release/mobile/android/base/java/org/mozilla/gecko/controlcenter/OverviewFragment.java]: as per Ravjit comment and implementation of safeGetBundle it won't return null, it creates empty bundle and fill it.
so you can check if the bunlde empty, and doesn't contain any keys using:
GeckoBundleUtils.safeGetBundle(controlCenterSettingsData, "data").isEmpty()

**Pull request merged**